lib/api/controllers/documentController.js

```diff
@@ -182,15 +181,16 @@ class DocumentController extends BaseController {
         modifiedRequest = newRequest;
 
         // @deprecated - Notifications before action will be removed in Kuzzle v2
-        this.kuzzle.notifier.publish(modifiedRequest, 'unknown', 'pending');
-
-        return this.engine.create(modifiedRequest);
+        return this.kuzzle.notifier.publish(
+          modifiedRequest, 'unknown', 'pending');
       })
-      .then(response => {
-        this.kuzzle.notifier.notifyDocumentCreate(modifiedRequest, response);
-
-        return response;
-      });
+      .then(() => this.engine.create(modifiedRequest))
+      .then(_response => {
+        response = _response;
```

Comment:
Why duplicate the response here ?

Reply:
because it's needed further on in the promises chain

Reply:
Oh yes sorry, I must be tired -_-